src/text-editor.js

```diff
@@ -4794,6 +4794,24 @@ class TextEditor {
           const indentLength = this.buffer.lineForRow(start).match(/^\s*/)[0].length
           this.buffer.insert([start, indentLength], commentStartString + ' ')
           this.buffer.insert([end, this.buffer.lineLengthForRow(end)], ' ' + commentEndString)
+
+          // Prevent the cursor from selecting / passing the delimiters
+          // See https://github.com/atom/atom/pull/17519
+          if (options.correctSelection && options.selection) {
+            let endLineLength = this.buffer.lineLengthForRow(end)
+            let oldRange = options.selection.getBufferRange()
+            if (oldRange.isEmpty()) {
+              if (oldRange.start.column === endLineLength) {
+                let endCol = endLineLength - commentEndString.length - 1
+                options.selection.setBufferRange([[end, endCol], [end, endCol]], {autoscroll: false})
+                return
```

Comment:
Maybe I'm missing something, but as far as I can tell this `return` statement is unnecessary.